src/xpk/core/ray.py

```diff
@@ -0,0 +1,134 @@
+"""
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from ..utils import write_tmp_file, xpk_print, xpk_exit
+from .commands import run_command_with_updates_retry
+
+
+RAY_VERSION = '2.34.0'
+
+ray_cluster_crd_yaml = """apiVersion: ray.io/v1
+kind: RayCluster
+metadata:
+  name: raycluster
+spec:
+  rayVersion: '{version}'
+  headGroupSpec:
+    rayStartParams: {{}}
+    #pod template
+    template:
+      spec:
+        containers:
+        - name: ray-head
+          image: rayproject/ray:{version}
+          resources:
+            limits:
+              cpu: 1
```

Comment:
These resources are probably insufficient for the ray head

Reply:
Added a mechanism to requisition memory and cpu as a percent of the available resources in the node. I set it to 50% for the head node, let me know what you think.